doc/ADR/ADR_001.md

```diff
@@ -0,0 +1,138 @@
+
+Thus, _State_ is inherently JSON-like, since we already know, from the design of Apalache's ITF, how to represent maps and arrays/vectors (representing structs/enums should be straightforward as well). 
+
+In contrast, _Interface_ is simply an unordered collection of the public methods admitted by a contract, and can be thought of a simple set or array.
+
+It is important to emphasize, that we do not need to capture function behavior in any way; the Transition Extractor observes a history of transactions that have already happened, and the effects of functions are observed as changes to the state of a contract, from one block to another.
+
+### Transition Extractor
+
+This component reads transaction data from the Soroban RPC, and builds a transaction log; an ordered sequence of smart contract States (or tuples of States, if we're observing a collection of contracts), annotated with the names of the functions called at each step.
+
+From a technical perspective, a key feature of the Transition Extractor, is dealing with the way the Soroban RPC serves data - the XDR format.
+
+There are several [RPC methods](RPC) avaliable, though two are of particular interest to us:
+
+	- `getLedgerEntries`, which allows us to probe any contract state, provided we can interpret the result XDR
+	- `getTransaction`, which allows us to determine which functions were called. 
+	
+Using `getLedgerEntries`, we should be able to construct a smart contract _State_ (as defined above) object, and avoid XDR in other components of Solarkraft. [This guide](RPCguide) should be used to get a general idea of the steps required here.
+
+#### Interface
+We envision the following methods, as part of the Transition Extractor interface:
+
+```
+fn readState(contractId: ContractId) -> State
+```
+
+### Monitor Executor
+
+The role of this component is to interact with Apalache. 
+We want to avoid implementing any form of TLA+ parsing in Solarkraft, so this component needs to solve the following problem:
+Given a TLA+ monitor specification, which references internal named TLA+ state variables, the Monitor Executor needs to be able to translate our smart contract State format to a series of constraints over the monitor variables.
+
+We can then perform the following steps:
+
+	1. Monitor Executor receives a sequence of States from Transition Extractor
+	2. Using that sequence, Monitor Executor generates a TLA+ spec, which `EXTENDS` (but doesn't read) the monitor specification. This extension contains the constraints derived from the input States.
+	3. Monitor Executor submits the new spec to the Apalache server for checking, receives ok/nok
+
+In the case of a property violation, it would be best to delegate the alarm responsibilities to a different component, as not to overload Monitor Executor.
+
+#### Interface
+We envision the following methods, as part of the Monitor Executor interface:
+
+```
+fn stateToTLA(state: State, monitorContractName: str) -> TlaRepr
```

Comment:
I think this comment deserves a more general discussion, because I think the real question being asked here is whether we want a monitor over a trace , or a monitor over a state as the MVP. The idea was, that the critical thing to have is how you turn one state into a set of constraints (which you can then use to generate finite-trace constraints for multiple states).

Reply:
But to do meaningful monitoring you'd need at least a pair of states / a trace of length 2?

Reply:
I don't understand why we are talking about an MVP in the architecture document. I thought we wanted to think through a solid architecture. If we want to go full agile, we don't need many ADRs and RFCs.